src/lexer.coffee

```diff
@@ -93,11 +95,21 @@ exports.Lexer = class Lexer
   # returns, etc. If we’re lexing literate CoffeeScript, strip external Markdown
   # by removing all lines that aren’t indented by at least four spaces or a tab.
   clean: (code) ->
-    code = code.slice(1) if code.charCodeAt(0) is BOM
-    code = code.replace(/\r/g, '').replace TRAILING_SPACES, ''
+    thusFar = 0
+    if code.charCodeAt(0) is BOM
+      code = code.slice 1
+      @locationDataCompensations[0] = 1
```

Comment:
The basic idea is that we have to record the "compensations" for where we stripped out characters at certain offsets in the source

So here record a compensation for an initial BOM